### PR TITLE
EICNET-2758: [Event] No one should be allowed to comment on a draft story

### PR DESCRIPTION
**Pull request by 4hardfire**
### Fixes

- Hide comments block and access to block api endpoints when content is in draft or unpublished state.

### Todo

- [x] Move methods from ContentModerationManager to EICModerationHelper after the following PR has been merged #2002

### Test

- [x] As SA/SCM, create a draft News in a global event
- [x] Make sure you cannot post or reply to comments
- [x] Publish the News and make sure you can post or reply to comments now
- [x] Unpublish the News and make sure you cannot post or reply to comments
- [x] Make sure the same behaviour happens with all group content. Whenever the content is in draft, users cannot post or reply to comments.

### CODE REVIEW

File: lib/modules/eic_groups/src/Controller/DiscussionController.php [hunk 1]
```diff
@@ -655,13 +752,15 @@ private function renderCommentElement(CommentInterface $comment): array {
...
+   *   The group content node ID.
    *
-   * @return bool
-   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
-   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @return GroupInterface|bool
+   *   The group entity.
    */
-  private function isGroupArchived(int $node_id): bool {
+  private function loadGroupFromNode(int $node_id): bool {
```
Comment (lramarojaona) [inline lib/modules/eic_groups/src/Controller/DiscussionController.php:763]: We already have a similar function in `Drupal\eic_groups\EICGroupsHelper::getGroupFromContent()`

**Review by lramarojaona — CHANGES_REQUESTED** [review]
It works, but I have 2 remarks.

File: lib/modules/eic_groups/src/Controller/DiscussionController.php (hunk 1 above)
Comment (4hardfire) [reply lib/modules/eic_groups/src/Controller/DiscussionController.php]: Done!

File: lib/modules/eic_groups/src/Plugin/Block/CommentsFromDiscussionBlock.php [hunk 2]
```diff
@@ -134,22 +142,24 @@ public static function create(
    *   The plugin_id for the plugin instance.
    * @param mixed $plugin_definition
    *   The plugin definition.
-   * @param \Drupal\eic_groups\EICGroupsHelper $groups_helper
+   * @param EICGroupsHelper $groups_helper
```
[the remark this answers is not in the document]
Comment (4hardfire) [reply lib/modules/eic_groups/src/Plugin/Block/CommentsFromDiscussionBlock.php]: Fixed!

**Review by lramarojaona — APPROVED** [review]
Ok, it works!

**Pull request merged**